src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,1591 @@
+			
+			bool DehumidificationRequested = false;
+			bool HumidificationRequested = false;
+			// Load required to meet dehumidifying setpoint (<0 = a dehumidify load)  [kgWater/s]
+			if (StepIns.ZoneDehumidificationLoad < 0) DehumidificationRequested = true;
+			// Load required to meet humidifying setpoint (>0 = a humidify load) [kgWater/s]
+			if (StepIns.ZoneMoistureLoad > 0) HumidificationRequested = true;
+			//RequestedHumdification = StepIns.ZoneMoistureLoad;
+			//RequestedDeHumdification = StepIns.ZoneDehumidificationLoad;
+			double averaged_requestedLoad = 0;
+
+			std::list<CMode*>::const_iterator iterator;
+			iterator = OperatingModes.begin();
+			++iterator;
+			for (; iterator != OperatingModes.end(); ++iterator) // iterate though the modes.
+			{
+				CMode* pMode = *iterator;
+				CModeSolutionSpace* solutionspace = &(pMode->sol);
+				bool SAHR_OC_MetinMode = false;
+				bool SAT_OC_MetinMode = false;
+				int solution_map_sizeX = solutionspace->PointX.size() - 1;
+				int solution_map_sizeY = solutionspace->PointY.size() - 1;
+				int solution_map_sizeM = solutionspace->PointMeta.size() - 1;
+				if (solution_map_sizeX != solution_map_sizeY)
+				{
+					ShowWarningError("Error in solution space mapping, suggest adjusting operating constraints.");
+					return -2;//error
+				}
+				// Check that in this mode the //Outside Air Relative Humidity(0 - 100 % )	//Outside Air Humidity Ratio(g / g)//Outside Air Temperature(�C)
+				if (pMode->MeetsOAEnvConstraints(StepIns.Tosa, Wosa, 100 * StepIns.RHosa) == true)//fix this it should not end if it does not meet contraints.
+				{
+					EnvironmentConditionsMet = EnvironmentConditionsMetOnce = true;
+				}
+				else
+				{
+					EnvironmentConditionsMet = false;
+					if (!WarmupFlag)
+					{
+						//ShowWarningError("Failed to meet environment conditions for mode: " + RoundSigDigits(double(pMode->ModeID), 1));
+					}
+				}
+
+				if (EnvironmentConditionsMet == true)
+				{
+					double NormalizationReference = GetNormalPoint(1);// assumes the model has at least 1 curve and that all are set the same.
+
+					for (point_number = 0; point_number != solution_map_sizeX; point_number++) // within each mode go though all the combinations of solution spaces.
+					{
+						//Supply Air Mass Flow Rate(kg / s)
+						//Outside Air Fraction(0 - 1)
+						CSetting* pCandidateSetting = new CSetting;
+						MsaRatio = solutionspace->PointX[point_number];// fractions of rated mass flow rate, so for some modes this might be low but others hi
+						OSAF = solutionspace->PointY[point_number];
+						// there is no operating condition test, becuase it uses those to make the map so only allowed values are here.
+						Msa = NormalizationReference * MsaRatio;
+						//Calculate the ventilation mass flow rate
+						Mvent = Msa * OSAF;
+
+						if (StdRhoAir>1) 	pCandidateSetting->Supply_Air_Ventilation_Volume = Mvent / StdRhoAir;
```

Comment:
This if convention is really hard to read, we prefer braces, even for single line blocks.  The positioning of the curly braces is somewhat up to discussion so I'm not worried about that for now, but we do want curly braces.